app/screens/home/search/results/file_options/file_options.tsx

```diff
@@ -0,0 +1,210 @@
+        },
+        imageVideo: {
+            marginLeft: 10,
+            height: 72,
+            width: 72,
+        },
+        optionContainer: {
+            marginHorizontal: -20,
+            paddingHorizontal: 20,
+        },
+        infoContainer: {
+            marginVertical: 8,
+            alignItems: 'center',
+            flexDirection: 'row',
+        },
+        infoText: {
+            flexDirection: 'row',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        toast: {
+            marginTop: 100,
+            alignItems: 'center',
+        },
+    };
+});
+
+type Props = {
+    canDownloadFiles: boolean;
+    enablePublicLink: boolean;
+    fileInfo: FileInfo;
+}
+const FileOptions = ({fileInfo, canDownloadFiles, enablePublicLink}: Props) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const intl = useIntl();
+    const ref = useRef<any>();
+    const serverUrl = useServerUrl();
+    const [action, setAction] = useState<GalleryAction>('none');
+
+    const galleryItem = {...fileInfo, type: 'image'} as GalleryItemType;
+    const size = getFormattedFileSize(fileInfo.size);
+
+    const handleDownload = useCallback(() => {
+        setAction('downloading');
+    }, []);
+
+    const handleCopyLink = useCallback(() => {
+        setAction('copying');
+    }, []);
+
+    const handlePermalink = useCallback(async () => {
+        showPermalink(serverUrl, '', fileInfo.post_id, intl);
+    }, [fileInfo]);
+
+    const renderIcon = () => {
+        switch (true) {
+            case isImage(fileInfo):
+                return (
+                    <View style={style.imageVideo}>
+                        <ImageFile
+                            file={fileInfo}
+                            forwardRef={ref}
+                            inViewPort={true}
+                            resizeMode={'cover'}
+                        />
+                    </View>
+                );
+            case isVideo(fileInfo):
+                return (
+                    <View style={style.imageVideo}>
+                        <VideoFile
+                            file={fileInfo}
+                            forwardRef={ref}
+                            resizeMode={'cover'}
+                            inViewPort={true}
+                            index={0}
+                            wrapperWidth={78}
+                        />
+                    </View>
+                );
+            default:
+                return (
+                    <FileIcon
+                        file={fileInfo}
+                        iconSize={72}
+                    />
+                );
+        }
+    };
+
+    const renderHeader = () => {
```

Comment:
to be extracted in its own file as component

Reply:
added to https://mattermost.atlassian.net/browse/MM-45617